lib/puppet/provider/package/gem.rb

```diff
@@ -1,8 +1,9 @@
 require 'puppet/provider/package'
+require 'puppet/provider/package_targetable'
```

Comment:
If you rename the targetable provider to be in `puppet/provider/package/targetable.rb` you might not need this require? The autoloader might find it in that case (though I'm not 100% sure of that)

Reply:
Not (auto)loaded with `require` when renamed to `package/targetable.rb:

```
mv provider/package_targetable.rb  provider/package/targetable.rb 
```

vi package/gem.rb

```
# require 'puppet/provider/package'
```

```
Error: Could not autoload puppet/provider/package/gem: uninitialized constant Puppet::Provider::Package::Targetable
```

Reply:
It would be nice for it to autoload!